src/opm/io/eclipse/ERst.cpp

```diff
@@ -20,46 +20,44 @@
 
 #include <algorithm>
 #include <cstring>
+#include <exception>
 #include <iomanip>
 #include <iterator>
-#include <string>
 #include <sstream>
+#include <stdexcept>
+#include <string>
 
-namespace Opm { namespace EclIO {
-
-ERst::ERst(const std::string& filename)
-    : EclFile(filename)
-{
-    loadData("SEQNUM");
+#include <boost/regex.hpp>
 
-    std::vector<int> firstIndex;
+namespace {
+    int seqnumFromSeparateFilename(const std::string& filename)
+    {
+        const auto re = boost::regex {
```

Comment:
I guess we still have to use `boost::regex` for the formal gcc version requirement (4.8.x??) - but for all "practical purposes" we have `std::regex` available. I know it is not for this PR - but maybe it is time to up the minimum compiler version?

Reply:
> I guess we still have to use `boost::regex` for the formal gcc version requirement (4.8.x??) - but for all "practical purposes" we have `std::regex` available

Just for the record, a full implementation of `std::regex` is available in GCC as of 4.9.x.

> I know it is not for this PR - but maybe it is time to up the minimum compiler version?

I'm sure we'll raise our minimum requirement before the next public release. At that point we'll probably require a compiler that's able to build Dune 2.6 (which GCC 4.9 isn't).